### PR TITLE
MAISTRA-770 only update status once at the end of reconcile

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

**Review by knrc — APPROVED** [review]

File: pkg/controller/servicemesh/controlplane/reconciler.go [hunk 1]
```diff
@@ -225,22 +231,40 @@ func (r *ControlPlaneReconciler) Reconcile() (result reconcile.Result, err error
...
+
 	r.Log.Info("Completed ServiceMeshControlPlane reconcilation")
-	// requeue to ensure readiness is updated
-	result.Requeue = true
 	return
 }
 
+func (r *ControlPlaneReconciler) pauseReconciliation(chartName string, err error) (string, error) {
+	var reason string
+	if r.isUpdating() {
```
Comment (brian-avery) [inline pkg/controller/servicemesh/controlplane/reconciler.go:252]: might be cleaner to say: 

reason := eventReasonInstall
if r.IsUpdating() {
  reason = eventReasonPausingUpdate
}

File: pkg/controller/servicemesh/controlplane/reconciler.go [hunk 2]
```diff
@@ -423,23 +447,46 @@ func (r *ControlPlaneReconciler) PostStatus() error {
 }
 
 func (r *ControlPlaneReconciler) postReconciliationStatus(reconciliationMessage string, processingErr error) error {
+	var reason string
```
Comment (brian-avery) [inline pkg/controller/servicemesh/controlplane/reconciler.go:450]: same as above

**Review by brian-avery — APPROVED** [review]

**Review by jwendell — APPROVED** [review]
Tested and it fixed the issue

File: pkg/controller/servicemesh/controlplane/reconciler.go (hunk 1 above)
Comment (rcernich) [reply pkg/controller/servicemesh/controlplane/reconciler.go:252]: Saves resetting the variable.  I guess it depends on preference.

**Pull request merged**